mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -1573,76 +1589,77 @@ pub struct ElectrumClientImpl {
 async fn electrum_request_multi(
     client: ElectrumClient,
     request: JsonRpcRequestEnum,
-) -> Result<(JsonRpcRemoteAddr, JsonRpcResponseEnum), String> {
+) -> Result<(JsonRpcRemoteAddr, JsonRpcResponseEnum), JsonRpcErrorType> {
     let mut futures = vec![];
     let connections = client.connections.lock().await;
     for (i, connection) in connections.iter().enumerate() {
         let connection_addr = connection.addr.clone();
-        match &*connection.tx.lock().await {
-            Some(tx) => {
-                let fut = electrum_request(
-                    request.clone(),
-                    tx.clone(),
-                    connection.responses.clone(),
-                    ELECTRUM_TIMEOUT / (connections.len() - i) as u64,
-                )
-                .map(|response| (JsonRpcRemoteAddr(connection_addr), response));
-                futures.push(fut)
-            },
-            None => (),
+        let json = json::to_string(&request).map_err(|e| JsonRpcErrorType::InvalidRequest(e.to_string()))?;
```

Comment:
Why is it required to serialize the request here?
`electrum_request` did it before.
It causes the `electrum_request` function to take one argument more than before.

Reply:
This was suggested by me here https://github.com/KomodoPlatform/atomicDEX-API/pull/1446#discussion_r961805246, I actually requested your opinion @sergeyboyko0791  on this :).
Do you have a better solution for the problem in the comment? or do you think my other suggestion in the comment was a better solution than serializing the request outside the `electrum_request` function?
Please also note that the `electrum_request` function should only have transport related code only from now on.

Reply:
Now I see the reason why `electrum_request` returns Transport errors **only**. I agree with this.
> This is not ideal, but it's better than matching the errors in the vector and returning JsonRpcErrorType::InvalidRequest or JsonRpcErrorType::Transport if all the errors are of the same kind, otherwise returning JsonRpcErrorType::Internal. What do you think @sergeyboyko0791 @borngraced ?

@borngraced please add `# Important` title to the doc comment for `electrum_request`.